src/coreclr/jit/emit.cpp

```diff
@@ -4102,6 +4127,198 @@ void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
+                assert(jmp == id);
+#endif // DEBUG
+
+                JITDUMP("IG%02u IN%04x is the last instruction in the group and jumps to the next instruction group IG%02u %s, removing.\n",
+                        jmpGroup->igNum, jmp->idDebugOnlyInfo()->idNum, targetGroup->igNum,
+                        emitLabelString(targetGroup));
+
+                // Unlink the jump from emitJumpList while keeping the previousJmp the same.
+                if (previousJmp != nullptr)
+                {
+                    previousJmp->idjNext = jmp->idjNext;
+                    if (jmp == emitJumpLast)
+                    {
+                        emitJumpLast = previousJmp;
+                    }
+                }
+                else
+                {
+                    assert(jmp == emitJumpList);
+                    emitJumpList = jmp->idjNext;
+                }
+
+                UNATIVE_OFFSET codeSize = jmp->idCodeSize();
+                jmp->idCodeSize(0);
+
+                jmpGroup->igSize -= (unsigned short)codeSize;
+                jmpGroup->igFlags |= IGF_UPD_ISZ;
+
+                emitTotalCodeSize -= codeSize;
+                totalRemovedSize += codeSize;
+                totalRemovedCount += 1;
+            }
+            else
+            {
+                // Update the previousJmp
+                previousJmp = jmp;
+#if DEBUG
+                if (targetGroup == nullptr)
+                {
+                    JITDUMP("IG%02u IN%04x jump target is not set!, keeping.\n",
+                            jmpGroup->igNum,jmp->idDebugOnlyInfo()->idNum );
+                }
+                else if ((jmpGroup->igFlags & IGF_HAS_ALIGN) != 0)
+                {
+                    JITDUMP("IG%02u IN%04x containing instruction group has alignment, keeping.\n",
+                            jmpGroup->igNum,jmp->idDebugOnlyInfo()->idNum);
+                }
+                else if (jmpGroup->igNext != targetGroup)
+                {
+                    JITDUMP("IG%02u IN%04x does not jump to the next instruction group, keeping.\n",
+                            jmpGroup->igNum,jmp->idDebugOnlyInfo()->idNum);
+                }
+#endif // DEBUG
+            }
+        }
+        else
+        {
+            // Update the previousJmp
+            previousJmp = jmp;
+        }
+
+        if (totalRemovedCount>0)
+        {
+            insGroup* adjOffIG     = jmpGroup->igNext;
+            insGroup* adjOffUptoIG = nextJmp != nullptr ? nextJmp->idjIG : emitIGlast;
+            while ((adjOffIG != nullptr) && (adjOffIG->igNum <= adjOffUptoIG->igNum))
+            {
+                JITDUMP("Adjusted offset of IG%02u from %04X to %04X\n", adjOffIG->igNum, adjOffIG->igOffs,
+                        (adjOffIG->igOffs - totalRemovedSize));
+                adjOffIG->igOffs -= totalRemovedSize;
+                adjOffIG = adjOffIG->igNext;
+            }
+        }
+
+        jmp = nextJmp;
+    }
+
+    if (totalRemovedCount > 0)
```

Comment:
Why can't this be `totalRemovedSize > 0`? Then, you won't need `totalRemovedCount`.

Reply:
It could be either but since I've got both so I can print them in debug mode i just went with the closest one, I really didn't think it mattered that much.

Reply:
I would recommend to get rid of it. Don't think it is helpful to see how many jumps we eliminated.

Reply:
count is now in DEBUG only, size alone is used in release.